python/packages/autogen-core/docs/src/user-guide/agentchat-user-guide/tutorial/messages.ipynb

```diff
@@ -0,0 +1,79 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Messages"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "In AutoGen, _messages_ facilitate communication and information exchange with other agents, orchestrators, and applications. AgentChat supports various message types, each designed for specific purposes."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Types of Messages\n",
+    "\n",
+    "At a high level, messages in AgentChat can be categorized into two types: agent-agent messages and an agent's internal events and messages.\n",
+    "\n",
+    "### Agent-Agent Messages\n",
+    "AgentChat supports many message types for agent-to-agent communication. The most common ones is the {py:class}`~autogen_agentchat.messages.ChatMessage`. This message type allows both text and multimodal communication and subsumes other message types, such as {py:class}`~autogen_agentchat.messages.TextMessage` or {py:class}`~autogen_agentchat.messages.MultiModalMessage`.\n",
+    "\n",
+    "For example, the following code snippet demonstrates how to create a text message, which accepts a string content and a string source:\n",
+    "\n",
+    "```python\n",
+    "from autogen_agentchat.messages import TextMessage\n",
+    "\n",
+    "message = TextMessage(content=\"Hello, world!\", source=\"User\")\n",
+    "\n",
+    "```\n",
+    "\n",
+    "Similarly, the following code snippet demonstrates how to create a multimodal message, which accepts\n",
+    "a list of strings or {py:class}`~autogen_core.Image` objects:\n",
+    "\n",
+    "```python\n",
+    "from autogen_agentchat.messages import MultiModalMessage\n",
+    "from autogen_core import Image\n",
+    "\n",
+    "img = Image(\"path/to/image.png\")\n",
+    "message = MultiModalMessage(content=[\"Hello, world!\", img)], source=\"User\")\n",
+    "\n",
+    "```\n",
+    "\n",
+    "You'll later find these message types useful when you want to send and receive messages from other agents.\n",
+    "\n",
+    "### Internal Events and Messages\n",
+    "Similarly, the module supports many message types for an agent's internal events and messages. These include {py:class}`~autogen_agentchat.messages.ToolCallMessage`, which is used to call tools, and {py:class}`~autogen_agentchat.messages.ToolCallResultMessage`, which contains the results of tool calls.\n",
+    "\n",
+    "For instance, suppose you want to call a tool with the name \"tool_name\" and the arguments \"arg1\" and \"arg2\". You can create a {py:class}`~autogen_agentchat.messages.ToolCallMessage` as follows:\n",
```

Comment:
I think it is important to note that the tool call messages are emitted for observability, rather than for controlling the agent. It is important to point out that these are the messages you want to emit when you implement a custom agent for observability.

Reply:
Currently the way it is written feels like you will pass a tool call message to an agent, and the agent will execute tool call -- it is not the case.

Reply:
I am taking a pass now, I'll try to fix this and other comments. 
There also seems to be a bug in the MultiModal message part which I will try to fix.

Reply:
Thank you Victor!